### PR TITLE
Add support for Python 3.14

### LINKED ISSUE #1960: Support for Python 3.14
Enable support for Python 3.14, also in the CI. Upgrade dependency packages as necessary.

### PR DESCRIPTION
**Pull request by leoll2**
### Summary

This PR enables compatibility with the latest Python 3.14, while dropping support for the old Python 3.10. Removing 3.10 was necessary because there's no numpy version which supports 3.10 and 3.14 at the same time. This PR also upgrades some dependency packages to a version compatible with 3.14; for breaking changes (e.g. in `shapely`) I have updated the source code.
CI was also updated to test both the minimum (3.11) and maximum (3.14) supported versions.

Resolves #1960 

### Checklist
- [x] I have added tests to cover my changes or documented any manual tests.
- [x] I have updated the [documentation](https://github.com/open-edge-platform/datumaro/tree/develop/docs) accordingly

### CODE REVIEW

File: src/datumaro/experimental/tiling/tilers.py [hunk 1]
```diff
@@ -326,9 +327,9 @@ def extract_tile(
         return pl.DataFrame(results)
 
 
-def _apply_offset(geom: sg.base.BaseGeometry, offset_x: float, offset_y: float) -> sg.base.BaseGeometry:
+def _apply_offset(geom: BaseGeometry, offset_x: float, offset_y: float) -> BaseGeometry:
     """Apply offset to geometry."""
-    return so.transform(lambda x, y: (x - offset_x, y - offset_y), geom)
+    return transform(geometry=geom, transformation=lambda x, y: (x - offset_x, y - offset_y), interleaved=False)
```
[the remark this answers is not in the document]
Comment (leoll2) [reply src/datumaro/experimental/tiling/tilers.py:332]: Wrong, the default is `True`. https://shapely.readthedocs.io/en/latest/reference/shapely.transform.html

**Comment by leoll2** [comment]
Right before merging this PR, we must update the required GHA workflows in the repo settings (from 3.10+3.13 to 3.11+3.14). I'll handle that.

File: .github/workflows/publish_to_pypi.yml [hunk 2]
```diff
@@ -53,10 +53,10 @@ jobs:
         uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
         with:
           persist-credentials: false
-      - name: Set up Python 3.10
+      - name: Set up Python
         uses: actions/setup-python@e797f83bcb11b83ae66e0230d6156d7c80228e7c # v6.0.0
         with:
-          python-version: "3.10"
+          python-version: "3.13"
```
Comment (AlbertvanHouten) [inline .github/workflows/publish_to_pypi.yml]: why not 3.11 or 3.14 here?

Comment (leoll2) [reply .github/workflows/publish_to_pypi.yml]: Python version doesn't really matter for this operation. I've just updated it to 3.14 anyway.

Comment (AlbertvanHouten) [reply .github/workflows/publish_to_pypi.yml]: I know, it will be slightly better since 3.11/3.14 should be cached already.

**Review by AlbertvanHouten — APPROVED** [review]
